python/lsst/meas/base/compensatedGaussian/_compensatedTophat.py

```diff
@@ -0,0 +1,215 @@
+        minLength=1,
+        default=[12,],
+    )
+    inner_scale = RangeField(
+        doc="Inner background annulus scale (relative to aperture).",
+        dtype=float,
+        default=1.0,
+        min=1.0,
+    )
+    outer_scale = RangeField(
+        doc="Outer background annulus scale (relative to aperture).",
+        dtype=float,
+        default=1.7,
+        min=1.0,
+    )
+
+    def validate(self):
+        super().validate()
+
+        if not (self.outer_scale > self.inner_scale):
+            raise ValueError("The outer_scale must be greater than the inner_scale")
+
+
+@register("base_CompensatedTophatFlux")
+class SingleFrameCompensatedTophatFluxPlugin(SingleFramePlugin):
+    ConfigClass = SingleFrameCompensatedTophatFluxConfig
+
+    @classmethod
+    def getExecutionOrder(cls):
+        return cls.FLUX_ORDER
+
+    def __init__(
+        self,
+        config: SingleFrameCompensatedTophatFluxConfig,
+        name: str,
+        schema,
+        metadata,
+        logName=None,
+        **kwds,
+    ):
+        super().__init__(config, name, schema, metadata, logName, **kwds)
+
+        flagDefs = FlagDefinitionList()
+
+        self.aperture_keys = {}
+        self._rads = {}
+        self._inner_scale = config.inner_scale
+        self._outer_scale = config.outer_scale
+        for aperture in config.apertures:
+            base_key = f"{name}_{aperture}"
+
+            # flux
+            flux_str = f"{base_key}_instFlux"
+            flux_key = schema.addField(
+                flux_str,
+                type="D",
+                doc="Compensated Tophat flux measurement.",
+                units="count",
+            )
+
+            # flux error
+            err_str = f"{base_key}_instFluxErr"
+            err_key = schema.addField(
+                err_str,
+                type="D",
+                doc="Compensated Tophat flux error.",
+                units="count",
+            )
+
+            # mask bits
+            mask_str = f"{base_key}_mask_bits"
+            mask_key = schema.addField(mask_str, type=np.int32, doc="Mask bits set within aperture.")
+
+            # failure flags
+            failure_flag = flagDefs.add(f"{aperture}_flag", "Compensated Tophat measurement failed")
+            oob_flag = flagDefs.add(f"{aperture}_flag_bounds", "Compensated Tophat out-of-bounds")
+
+            self.aperture_keys[aperture] = (flux_key, err_key, mask_key, failure_flag, oob_flag)
+            self._rads[aperture] = int(math.ceil(self._outer_scale*aperture))
+
+        self.flagHandler = FlagHandler.addFields(schema, name, flagDefs)
+        self._max_rad = max(self._rads)
+
+    def fail(self, measRecord, error=None):
+        if error is None:
+            self.flagHandler.handleFailure(measRecord)
+        else:
+            self.flagHandler.handleFailure(measRecord, error.cpp)
+
+    def measure(self, measRecord, exposure):
+        center = measRecord.getCentroid()
+        bbox = exposure.getBBox()
```

Comment:
I forgot to post this comment earlier. Should we check if `bbox.getArea() > 0`. Many plugins have this check and would produce warnings for empty footprints.

Reply:
This ship has already sailed, but fortunately this isn't a problem.  If the bbox is empty then the center will never be inside the bbox so the `oob_flag` check below will trigger.